applications/ContactStructuralMechanicsApplication/custom_strategies/custom_builder_and_solvers/contact_residualbased_block_builder_and_solver.h

```diff
@@ -0,0 +1,342 @@
+        ModelPart& rModelPart,
+        TSystemVectorType& b
+        ) override
+    {
+        FixIsolatedNodes(rModelPart);
+        
+        BaseType::BuildRHS(pScheme, rModelPart, b);
+        
+        FreeIsolatedNodes(rModelPart);
+    }
+    
+    ///@}
+    ///@name Operations
+    ///@{
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+    ///@}
+
+private:
+    ///@name Static Member Variables 
+    ///@{
+
+    ///@}
+    ///@name Member Variables 
+    ///@{
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+    /**
+     * This method check the ISOLATED nodes and it fixes
+     * @param rModelPart The model part to compute
+     */
+    void FixIsolatedNodes(ModelPart& rModelPart)
+    {
+        KRATOS_ERROR_IF(!(rModelPart.HasSubModelPart("ComputingContact"))) << "ERROR:: CONTACT COMPUTING MODEL PART NOT CREATED" << std::endl;
+        ModelPart& computing_contact_model_part = rModelPart.GetSubModelPart("ComputingContact"); 
+        
+        // We reset the flag
+        auto& nodes_array = computing_contact_model_part.Nodes();
+    #ifdef _OPENMP
+        #pragma omp parallel for 
+    #endif
+        for(int i = 0; i < static_cast<int>(nodes_array.size()); ++i)
+            (nodes_array.begin() + i)->Set(ISOLATED, false);
+        
+        // Now we set the flag in the nodes
+        auto& conditions_array = computing_contact_model_part.Conditions();
+        
+    #ifdef _OPENMP
+        #pragma omp parallel for 
+    #endif
+        for(int i = 0; i < static_cast<int>(conditions_array.size()); ++i) 
+        {
+            auto cond_it = conditions_array.begin() + i;
+            
+            auto& geom = cond_it->GetGeometry();
+            for (std::size_t i_node = 0; i_node < geom.size(); ++i_node)
+            {
+                geom[i_node].SetLock();
+                geom[i_node].Set(ISOLATED, cond_it->Is(ISOLATED));
+                geom[i_node].UnSetLock();
+            }
+        }
+        
+        // We fix the LM
+    #ifdef _OPENMP
+        #pragma omp parallel for 
+    #endif
+        for(int i = 0; i < static_cast<int>(nodes_array.size()); ++i)
+        {
+            auto node_it = nodes_array.begin() + i;
+            if (node_it->Is(ISOLATED) == true)
+            {
+                auto& dofs = node_it->GetDofs();
+                
+                for (auto it_dof = dofs.begin(); it_dof != dofs.end(); ++it_dof)
+                {
+                    if (it_dof->IsFree())

```

Comment:
i am not sure of why you check if it is free.

however the clean version of what you are doing would be 

     if(node_it->IsFree(DISPLACEMENT_X)
         node_it->Fix(DISPLACEMENT_X;

i would do it without the first check though...overall this would imply

     if (node_it->Is(ISOLATED) == true)
     {
          node_it->Fix(DISPLACEMENT_X);
          node_it->Fix(DISPLACEMENT_Y);
          node_it->Fix(DISPLACEMENT_Z);
     }

Reply:
I don't want to fix displacements, I want to fix the LM. It is unilateral contact, then the movement is free but the LM are not computed